geo/src/algorithm/centroid.rs

```diff
@@ -454,6 +444,25 @@ where
     }
 }
 
+fn centroid_of_coords<T>(
+    mut coords_iter: impl Iterator<Item = Coordinate<T>>,
```

Comment:
Looks great as-is, though if you wanted you could use `impl ExactSizeIterator` and use `len()` instead of calculating length yourself

Reply:
alas:
```
error[E0277]: the trait bound `FlatMap<impl Iterator, Option<geo_types::Coordinate<T>>, [closure@geo/src/algorithm/centroid.rs:180:35: 180:78]>: ExactSizeIterator` is not satisfied
   --> geo/src/algorithm/centroid.rs:179:21
    |
179 | /                     self.iter()
180 | |                         .flat_map(|line_string| Some(*line_string.0.first()?)),
    | |______________________________________________________________________________^ the trait `ExactSizeIterator` is not implemented for `FlatMap<impl Iterator, Option<geo_types::Coordinate<T>>, [closure@geo/src/algorithm/centroid.rs:180:35: 180:78]>`
...
447 |   fn centroid_of_coords<T>(
    |      ------------------ required by a bound in this
448 |       mut coords_iter: impl ExactSizeIterator<Item = Coordinate<T>>,
    |                             --------------------------------------- required by this bound in `centroid_of_coords`

```

Reply:
Oh yeah we've got a flat map in there. Nvm